src/org/labkey/remoteapi/assay/Protocol.java

```diff
@@ -0,0 +1,355 @@
+        }
+    }
+
+    public JSONObject toJSONObject()
+    {
+        JSONObject result = new JSONObject();
+        result.put("protocolId", _protocolId);
+        result.put("name", _name);
+        result.put("description", _description);
+        result.put("providerName", _providerName);
+        JSONArray domains = new JSONArray();
+        result.put("domains", domains);
+        for (Domain domain : _domains)
+            domains.add(domain.toJSONObject(true));
+
+
+        if (_backgroundUpload != null)
+            result.put("backgroundUpload", _backgroundUpload);
+
+        if (_editableResults != null)
+            result.put("editableResults", _editableResults);
+
+        if (_qcEnabled != null)
+            result.put("qcEnabled", _qcEnabled);
+
+        if (_allowTransformationScript != null)
+            result.put("allowTransformationScript", _allowTransformationScript);
+        if (_saveScriptFiles != null)
+            result.put("saveScriptFiles", _saveScriptFiles);
+        if (_editableRuns != null)
+            result.put("editableRuns", _editableRuns);
+        if (_autoCopyTargetContainerId != null)
+            result.put("autoCopyTargetContainerId", _autoCopyTargetContainerId);
+
+        if (_selectedDetectionMethod != null)
+            result.put("selectedDetectionMethod", _selectedDetectionMethod);
+
+        if (_selectedMetadataInputFormat != null)
+            result.put("selectedMetadataInputFormat", _selectedMetadataInputFormat);
+
+        if (_selectedPlateTemplate != null)
+            result.put("seletedPlateTemplate", _selectedPlateTemplate);
+
+        if (_protocolParameters != null)
+            result.put("protocolParameters", _protocolParameters);
+
+        return result;
+    }
+
+    public Long getProtocolId()
+    {
+        return _protocolId;
+    }
+
+    public String getName()
+    {
+        return _name;
+    }
+
+    public Protocol setName(String name)
+    {
+        _name = name;
+        return this;
+    }
+
+    public String getDescription()
+    {
+        return _description;
+    }
+
+    public Protocol setDescription(String description)
+    {
+        _description = description;
+        return this;
+    }
+
+    public String getProviderName()
+    {
+        return _providerName;
+    }
+
+    public Protocol setProviderName(String providerName)
+    {
+        _providerName = providerName;
+        return this;
+    }
+
+    public List<Domain> getDomains()
+    {
+        return _domains;
+    }
+
+    public Protocol setDomains(List<Domain> domains)
+    {
+        _domains = domains;
+        return this;
+    }
+
+    public Boolean getAllowBackgroundUpload()
+    {
+        return _allowBackgroundUpload;
+    }
+
+    public Boolean getAllowEditableResults()
+    {
+        return _allowEditableResults;
+    }
+
+    public Protocol allowQCStates(boolean qcEnabled)
+    {
+        _allowQCStates = qcEnabled;
```

Comment:
This should be `setQcEnabled` and should be `_qcEnabled = qcEnabled` I believe.

Reply:
Oh nevermind, I see the setQcEnabled below, so this allowQcState setter should be removed.

Reply:
good catch.  Removing it